### DISCUSSION TITLE
VMware SVGA/SVGA II and VBoxVGA "emulation"

### DISCUSSION THREAD

**Discussion by Torinde** [86Box/86Box#2910] [Ideas]
VMware SVGA/SVGA II and VirtualBox VBoxVGA have drivers for OSes supported by 86box, so if those are emulated, then this will allow to run 3D games and programs that require higher levels of Direct3D/OpenGL and higher performance (using OpenGL pass-through to the host). Accuracy is maintained since the originals are also virtualization instances.

Background: joncampbell123/dosbox-x/issues/3405, #4573

**Comment by MrPepka** [comment]
Out of scope of 86box. The purpose of this program is to accurately emulate computer hardware, not to emulate some virtual graphics cards, for this you can use QEMU, VMWare etc

**Reply by Torinde** [reply]
QEMU, VMware and the regular virtualization solutions in general aren't so well equipped with emulating the rest of the old hardware.
So, I thought it'll be useful for 86box to be able to run games and programs requiring strong 3D performance.

**Reply by MrPepka** [reply]
The purpose of 86box is primarily to emulate a computer (processor, video card, sound card, hard drives, motherboards and other computer components) in order to preserve computer hardware, so IMO I don't see the point of emulating virtual graphics cards, especially since when using OpenGL from the host everything will depend on the graphics card and its driver in the host, so if someone has a GPU that does not support old versions of OpenGL, these virtual graphics cards and OpenGL passthrough will be of no use to him. In addition, for most 3D games (also from the early 2000s), you will probably need to emulate PIII and PIV anyway, and emulation of these CPUs will be very inefficient on today's computers (unless you have the latest Ryzen)

**Reply by Torinde** [reply]
I understand the purpose of preserving computer hardware via accurate emulation. At the same time - isn't that closely linked to the purpose of preserving/using the _software_ running on that hardware?
- 86box already supports as system the "Virtual PC 2007", similar to VMware/VirtualBox. 
- some games and programs require more 3D power/features than can be expected in realistic timeframes via software emulation.
Thus it makes sense to allow the _option_ to use host-accelerated 3D instead

~'missing PIII and PIV that are needed anyway to play the games that need 3D host-acceleration' - I think current GPU emulation speed is insufficient for a broader/older set of games, but even if we touch upon later ones only - as you mention, latest host CPUs are close to being sufficient to cover the CPU emulation part.

~'host GPU that does not support old versions of OpenGL' - I think both the VMware/VirtualBox 3D drivers and [QEMU-3dfx](https://github.com/kjliew/qemu-3dfx) use on the host-side the latest OpenGL (or maybe Vulkan/D3D?) translating to those the older APIs from the guest? @javispedro @kjliew

**Comment by pankoza2-pl** [comment]
We have Bochs SVGA now, which includes the VBIOS version used by VirtualBox too, but is 2D only, and can only use Generic VESA drivers

**Comment by Carmina16** [comment]
So would it be possible to add VMSVGA adapter? It's open-source AFAIK.
